maestro_cfg_test

```diff
@@ -0,0 +1,369 @@
+#!/usr/bin/env python3
+
+# Synopsis: Test configuration for 2 levels of aggregators with 8 samplers
+# First level has 3 aggregators, 2nd level has 1 aggregator
+
+import argparse
+import json
+import logging
+import os
+import TADA
+import tempfile
+import time
+import sys
+
+from distutils.spawn import find_executable
+from LDMS_Test import LDMSDCluster, LDMSDContainer, process_args, \
+                      add_common_args, jprint, parse_ldms_ls, cs_rm
+
+
+if __name__ != "__main__":
+    raise RuntimeError("This should not be imported as a module.")
+
+class Debug(object):
+    pass
+D = Debug()
+
+logging.basicConfig(format = "%(asctime)s %(name)s %(levelname)s %(message)s",
+                    level = logging.INFO)
+
+log = logging.getLogger(__name__)
+
+exec(open(os.getenv("PYTHONSTARTUP", "/dev/null")).read())
+
+#### default values #### ---------------------------------------------
+sbin_ldmsd = find_executable("ldmsd")
+if sbin_ldmsd:
+    default_prefix, a, b = sbin_ldmsd.rsplit('/', 2)
+else:
+    default_prefix = "/opt/ovis"
+
+#### argument parsing #### -------------------------------------------
+DESC = "Test for ldmsd configuration with maestro_ctrl"
+ap = argparse.ArgumentParser(description = "Configuration tests for agg with 2 samplers")
+add_common_args(ap)
+args = ap.parse_args()
+process_args(args)
+
+#### config variables #### -------------------------------------------
+USER = args.user
+PREFIX = args.prefix
+COMMIT_ID = args.commit_id
+SRC = args.src
+CLUSTERNAME = args.clustername
+DB = args.data_root
+NUM_COMPUTE = 8
+STORE_ROOT = "/store" # path inside container (aggs)
+CLUSTER_CONFIG = "etcd.yaml"
+CFG_PREFIX = "headnode"
+AGG_PORT = 10010
+SMPLR_PORT = 10002
+HEADNODE_PORT = 10104
+STORE_PATH = "/DATA15/orion"
+LDMS_CFG_FILE = "/opt/ovis/ldms_cfg.yaml"
+ETCD_FILE = "/opt/ovis/etcd.yaml"
```

Comment:
I think `STORE_PATH` should be based on `DB` (which could be specified via --data-root in the CLI). My system does not have "/DATA15/orion" directory.

Also, can `LDMS_CFG_FILE` and `ETCD_FILE` be in the `DB` directory? People running the test may not have write permission to `/opt/ovis` directory on the host, or their ovis installation (on the host, not container) may be in different path.

Reply:
Whoop, good catch on the STORE_PATH, and sure, good point. I'll change over LDMS_CFG_FILE and ETCD_FILE to be in DB directory as well.